coding-concepts.js

```diff
@@ -6,49 +6,49 @@
+// const colors = ["tangerine", "magenta", "lilac", "daffodil"]
 // console.log(colors.push("indigo"))
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: This will log an array with with 5 string elements
+// // b) Verify and explain: .push() is adding the string "indigo" to the end of the colors array.
 
-// --------------------1) What will this log?
+// // --------------------1) What will this log?
 
-const cohort = "LEARN 2023"
+// const cohort = "LEARN 2023"
 // console.log(cohort.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: This will log the character length of the variable declared cohort.
+// // b) Verify and explain: .length property when expressed to the string data type returns the number of characters in the string.
 
-// --------------------2) What will this log?
+// // --------------------2) What will this log?
 
-const greeting = "Hello World!"
+// const greeting = "Hello World!"
 // console.log(greeting[4])
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: This will log the 5th character in the string. [4]---> "o"
+// // b) Verify and explain: When evaluating characters in a string data type, the count start index 0.
 
-// --------------------3) What will this log?
+// // --------------------3) What will this log?
 
-const languages = ["JavaScript", "Ruby", "Python", "C++"]
-const index = 1
+// const languages = ["JavaScript", "Ruby", "Python", "C++"]
+// const index = 1
 // console.log(languages[index])
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: This will log the "Ruby"
+// // b) Verify and explain: console.log function is taking the languages array and passing in the index variable which has the value 1.  This will evaulate the 2nd index of the languages array.
 
-// --------------------4) What will this log?
+// // --------------------4) What will this log?
 
-const weekendDays = ["saturday", "sunday"]
-// console.log(weekendDays.toUpperCase())
+// const weekendDays = ["saturday", "sunday"]
+// // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: This will mutate the string element within the weekendDays array, and convert all the characters in each string element to uppercase letters.
+// // b) Verify and explain: .toUpperCase built in method can only mutate string  because of the properties unique to this data type
 
-// --------------------5) What will this log?
+// // --------------------5) What will this log?
 
-const dataTypes = ["number", "string", "Boolean", "undefined"]
+// const dataTypes = ["number", "string", "Boolean", "undefined"]
 // console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will log a number data type.
+// b) Verify and explain: typeof accessor will evaluate the dataTypes array, and return the number of elements within because of the .length property.
```

Comment:
In this case, length is the one returning a 4 for the array. That 4 is then evaluated by typeof and we get number.